p2p/src/peer_manager/peerdb/mod.rs

```diff
@@ -131,11 +131,28 @@ impl<T: NetworkingService> PeerDb<T> {
 
     /// Checks if the given address is already connected.
     pub fn is_address_connected(&self, address: &T::Address) -> bool {
-        self.addresses.contains(address)
+        self.connected_addresses.contains(address)
     }
 
-    pub fn known_addresses(&self) -> Vec<PeerAddress> {
-        self.addresses.iter().map(TransportAddress::as_peer_address).collect()
+    /// Selects requested count of peer addresses from the DB randomly.
+    /// Result could be shared with remote peers over network.
+    pub fn random_known_addresses(&self, count: usize) -> Vec<T::Address> {
+        // TODO: Use something more efficient (without iterating over the all addresses first)
+        let all_addresses = self.known_addresses.iter().cloned().collect::<Vec<_>>();
+        all_addresses
+            .choose_multiple(&mut make_pseudo_rng(), count)
+            .cloned()
+            .collect::<Vec<_>>()
+    }
+
+    /// Selects requested count of connected peer ids randomly.
```

Comment:
This isn't really what I meant. Please add what purpose this serves.

Reply:
Updated one more time.